File: solutions/cpp/grade-school/3/grade_school.cpp
#include <algorithm>
#include "grade_school.h"

using namespace grade_school;

const roster_t &school::roster() const {
    return roster_;
}

void school::add(std::string_view name, int grade) {
    auto &students = roster_[grade];
    students.insert(
            std::upper_bound(students.begin(), students.end(), name),
            std::string(name)
    );
}

students_t school::grade(int grade) const {
    auto it = roster_.find(grade);
    if (it == roster_.end()) {
        return {};
    }
    return it->second;
}

File: solutions/swift/grade-school/2/Sources/GradeSchool/GradeSchool.swift
class GradeSchool {
  var r: [Int: [String]] = [:]

  func addStudent(_ name: String, grade: Int) -> Bool {
    guard !roster().contains(name) else { return false }

    r[grade, default: []].append(name)
    return true
  }

  func studentsInGrade(_ grade: Int) -> [String] { r[grade]?.sorted() ?? [] }
  func roster() -> [String] { r.keys.sorted().flatMap(studentsInGrade) }
}
